source/yats/include/yats/task_container.h

```diff
@@ -14,8 +15,8 @@ namespace yats
 class abstract_task_container
 {
 public:
-    abstract_task_container(std::set<size_t> following_nodes)
-        : m_following_nodes(std::move(following_nodes))
+    abstract_task_container(const std::set<size_t> &following_nodes)
```

Comment:
the `&` is on the side of the type in all other cases. we should check how we do this throughout the project and include this in our formatting rules.